### ISSUE TITLE
Recreate the `files` folder symlinks on folder rename

### ISSUE THREAD

**Issue by importer** [contao/core-bundle#83]
> <a href="https://github.com/xchs"><img src="https://avatars.githubusercontent.com/u/1218766?v=3" align="left" width="42" height="42" hspace="10"></img></a> [Issue](https://github.com/contao/contao/issues/5) by @xchs
Saturday Jun 14, 2014 at 02:02 GMT

It could be reasonable to automatically update the `files` folder symlinks once the symlink target folder is renamed.

As it is now, we would always have to run the maintenance job to recreate the respective symlinks again:
* Create a new folder `foo` in the _"File manager"_ and enable the _"Public"_ attribute to make it become publicly accessible. This should create the corresponding symlink (` foo -> ../../files/foo`) in the `web/files` directory.
* Now, rename the folder `foo` to e.g. `bar`
* In the `web/files` directory there is still a symlink `foo` that points to the target folder `../../files/foo`, which, however, in the meantime no longer exists.
* Finally, run the _"Recreate the symlinks"_ maintenance job to update the symlink accordingly

In short: similar to the DBAFS synchronization it might come in handy to perform immediately a symlink update job as well.

**Comment by importer** [comment]
> <a href="https://github.com/leofeyer"><img src="https://avatars.githubusercontent.com/u/1192057?v=3" align="left" width="42" height="42" hspace="10"></img></a> Comment by @leofeyer
Saturday Jun 14, 2014 at 15:42 GMT

Custom upload folder names are not supported at all yet, so we should definitely take care of it.

**Comment by importer** [comment]
> <a href="https://github.com/leofeyer"><img src="https://avatars.githubusercontent.com/u/1192057?v=3" align="left" width="42" height="42" hspace="10"></img></a> Comment by @leofeyer
Sunday Jul 06, 2014 at 18:24 GMT

I have fixed the issue with the custom directory name in 9b40ccbfb86b39f629ff6573736138e22bd26d48.

However, I don't think we should automatically execute any magic such as synchronizing the file system with the database or recreating the symlinks when the folder name is changed. What if you e.g. accidentally type in the wrong name?

Since renaming the upload folder is typically something which will only be done once when an installation is set up or maybe after a major version update, I think it is a reasonable burden to require people to run the maintenance jobs manually – in the order and at the time they prefer to.

**Comment by importer** [comment]
> <a href="https://github.com/xchs"><img src="https://avatars.githubusercontent.com/u/1218766?v=3" align="left" width="42" height="42" hspace="10"></img></a> Comment by @xchs
Sunday Jul 06, 2014 at 21:29 GMT

I just wanted to try it out but unfortunately I get the following error message when I run `composer install --prefer-dist`:

![error](https://cloud.githubusercontent.com/assets/1218766/3490014/84dc9074-0554-11e4-9b2a-4ba8127873f2.png)

**Comment by importer** [comment]
> <a href="https://github.com/leofeyer"><img src="https://avatars.githubusercontent.com/u/1192057?v=3" align="left" width="42" height="42" hspace="10"></img></a> Comment by @leofeyer
Monday Jul 07, 2014 at 06:47 GMT

Did you really run `composer install`? Because the error message says the **post-update-cmd** event returned with an error.

**Comment by importer** [comment]
> <a href="https://github.com/xchs"><img src="https://avatars.githubusercontent.com/u/1218766?v=3" align="left" width="42" height="42" hspace="10"></img></a> Comment by @xchs
Monday Jul 07, 2014 at 10:23 GMT

Yeah, you are right. I did try `composer install --prefer-dist` as well as `composer update --prefer-dist`—both were terminated with the given error message.

A more detailed explanation (sorry for OT):
* Clone the repository into current directory: `$ git clone https://github.com/contao/contao.git .`
* Checkout the `develop` branch: `$ git checkout -t origin/develop`
* Run the `composer install --prefer-dist` command to get all the dependencies
* That, however, ended up with:
  ```
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for symfony/icu v1.2.1 -> satisfiable by symfony/icu[v1.2.1].
    - symfony/icu v1.2.1 requires lib-icu >=4.4 -> the requested linked library icu has the wrong version installed or is missing from your system, make sure to have the extension providing it.
  Problem 2
    - symfony/icu v1.2.1 requires lib-icu >=4.4 -> the requested linked library icu has the wrong version installed or is missing from your system, make sure to have the extension providing it.
    - symfony/intl v2.5.0 requires symfony/icu ~1.0-RC -> satisfiable by symfony/icu[v1.2.1].
    - Installation request for symfony/intl v2.5.0 -> satisfiable by symfony/intl[v2.5.0].
  ```

So, next try was to run `composer update --prefer-dist` instead of `composer install --prefer-dist`:
```
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Installing contao/idna (dev-master 176e754)
    Loading from cache

  - Installing contao/php-diff (dev-master 9928f73)
    Loading from cache

  - Installing contao/tcpdf (dev-master d0a2b26)
    Loading from cache

  - Installing contao/contao-bundle (dev-develop 2651af5)
    Loading from cache

  - Installing contao/contao-library (dev-develop ca3f742)
    Loading from cache

  - Installing contao-community-alliance/composer-plugin (2.2)
    Loading from cache

  - Installing contao/module-calendar (dev-develop 6414cdb)
    Loading from cache

  - installed 216 files
  - Installing contao/module-comments (dev-develop 3823594)
    Loading from cache

  - installed 136 files
  - Installing contao/module-core (dev-develop 66cf932)
    Loading from cache

  - installed 913 files
  - Installing contao/module-devtools (dev-develop 8190b5d)
    Loading from cache

  - installed 121 files
  - Installing contao/module-faq (dev-develop 8d760cb)
    Loading from cache

  - installed 180 files
  - Installing contao/module-listing (dev-develop ecab132)
    Loading from cache

  - installed 53 files
  - Installing contao/module-news (dev-develop cf5b159)
    Loading from cache

  - installed 240 files
  - Installing contao/module-newsletter (dev-develop c5ae030)
    Loading from cache

  - installed 232 files
  - Installing contao/module-repository (dev-develop 516af54)
    Loading from cache

  - installed 119 files
  - Installing simplepie/simplepie (1.3.1)
    Loading from cache

  - Installing leafo/scssphp (dev-master ff76df3)
    Loading from cache

  - Installing leafo/scssphp-compass (dev-master f1631dd)
    Loading from cache

  - Installing leafo/lessphp (v0.4.0)
    Loading from cache

  - Installing symfony/console (v2.5.0)
    Loading from cache

  - Installing symfony/finder (v2.5.0)
    Loading from cache

  - Installing psr/log (1.0.0)
    Loading from cache

  - Installing symfony/debug (v2.5.0)
    Loading from cache

  - Installing swiftmailer/swiftmailer (v5.2.1)
    Loading from cache

  - Installing michelf/php-markdown (1.4.1)
    Loading from cache

  - Installing symfony/yaml (v2.5.0)
    Loading from cache

  - Installing symfony/translation (v2.5.0)
    Loading from cache

  - Installing symfony/validator (v2.5.0)
    Loading from cache

  - Installing symfony/property-access (v2.5.0)
    Loading from cache

  - Installing symfony/process (v2.5.0)
    Loading from cache

  - Installing symfony/filesystem (v2.5.0)
    Loading from cache

  - Installing symfony/http-foundation (v2.5.0)
    Loading from cache

  - Installing symfony/event-dispatcher (v2.5.0)
    Loading from cache

  - Installing symfony/http-kernel (v2.5.0)
    Loading from cache

  - Installing symfony/options-resolver (v2.5.0)
    Loading from cache

  - Installing symfony/intl (v2.5.0)
    Loading from cache

  - Installing symfony/icu (v1.1.1)
    Loading from cache

  - Installing symfony/form (v2.5.0)
    Loading from cache

  - Installing symfony/dependency-injection (v2.5.0)
    Loading from cache

  - Installing symfony/config (v2.5.0)
    Loading from cache

  - Installing symfony/class-loader (v2.5.0)
    Loading from cache

  - Installing doctrine/lexer (v1.0)
    Loading from cache

  - Installing doctrine/annotations (v1.2.0)
    Loading from cache

  - Installing symfony/templating (v2.5.0)
    Loading from cache

  - Installing symfony/stopwatch (v2.5.0)
    Loading from cache

  - Installing symfony/security-core (v2.5.0)
    Loading from cache

  - Installing symfony/security-csrf (v2.5.0)
    Loading from cache

  - Installing symfony/routing (v2.5.0)
    Loading from cache

  - Installing symfony/framework-bundle (v2.5.0)
    Loading from cache

  - Installing sensio/distribution-bundle (v3.0.1)
    Loading from cache

symfony/validator suggests installing doctrine/cache (For using the default cached annotation reader and metadata cache.)
symfony/validator suggests installing egulias/email-validator (Strict (RFC compliant) email validation)
symfony/validator suggests installing symfony/expression-language (For using the 2.4 Expression validator)
symfony/http-kernel suggests installing symfony/browser-kit ()
symfony/form suggests installing symfony/twig-bridge (For templating with Twig.)
symfony/dependency-injection suggests installing symfony/proxy-manager-bridge (Generate service proxies to lazy load them)
symfony/security-core suggests installing symfony/expression-language (For using the expression voter)
symfony/security-core suggests installing ircmaxell/password-compat (For using the BCrypt password encoder in PHP <5.5)
symfony/routing suggests installing symfony/expression-language (For using expression matching)
symfony/framework-bundle suggests installing doctrine/cache (For using alternative cache drivers)
Writing lock file
Generating autoload files
Script system/bin/automator generateSymlinks handling the post-update-cmd event returned with an error



  [RuntimeException]
  Error Output:



update [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-plugins] [--no-custom-installers] [--no-scripts] [--no-progress] [--with-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [packages1] ... [packagesN]
```

Only when I remove (just for testing purposes) both the `post-install-cmd` and `post-update-cmd` from the `composer.json` and run again the `composer update --prefer-dist` command, all dependencies are resolved and the autoload files are created successfully. After that, I did run manually the automator job `$ system/bin/automator generateSymlinks` what has now been completed without any error message.

**Comment by importer** [comment]
> <a href="https://github.com/leofeyer"><img src="https://avatars.githubusercontent.com/u/1192057?v=3" align="left" width="42" height="42" hspace="10"></img></a> Comment by @leofeyer
Tuesday Jul 08, 2014 at 14:22 GMT

This seems to be #12 rather than a symlink or Automator issue.
